.github/workflows/build-auth.yaml

```diff
@@ -0,0 +1,54 @@
+name: Build Auth
+
+on:
+  workflow_call:
+    inputs:
+      dry-run:
+        description: "An indication of whether to commit/publish results"
+        required: true
+        type: string
+      is-dependabot:
+        description: "An indication of a dependabot pull request"
+        required: true
+        type: string
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
This is used a lot, is it really necessary?
I would expect a single checkout the begin with.

Reply:
It is necessary. each job is running in its own environment. they all need to check out the code, because they boot up, with no files.